src/test/java/com/github/fppt/jedismock/datastructures/streams/StreamIdTests.java

```diff
@@ -0,0 +1,259 @@
+package com.github.fppt.jedismock.datastructures.streams;
+
+import com.github.fppt.jedismock.exception.WrongStreamKeyException;
+import org.junit.jupiter.api.Test;
+
+import static java.lang.Long.toUnsignedString;
+import static org.junit.jupiter.api.Assertions.*;
+
+
+public class StreamIdTests {
+    @Test
+    void zeroComparisonWithZeroTest() {
+        StreamId zero = new StreamId(0, 0);
+        assertThrows(
+                WrongStreamKeyException.class,
+                zero::compareToZero,
+                "ERR The ID specified in XADD must be greater than 0-0"
+        );
+    }
+
+    @Test
+    void zeroComparisonWithPositiveKeysTest() {
+        StreamId other = new StreamId(0, 1);
+        assertDoesNotThrow(() -> assertSame(other, other.compareToZero()));
```

Comment:
no need in `assertDoesNotThrow`. If you're worrying about unchecked exception, just declare it in a test method, and test will fail if exception is thrown. The use case for `assertDoesNotThrow` is for usage in  [soft asserts](https://eliasnogueira.com/soft-asserts-why-should-you-use-them-for-unit-and-integration-tests/), and here we don't use soft asserts.

Anyway, this class should be rewritten to use AssertJ

Reply:
WrongStreamKeyException is a checked exception

Reply:
Yep it was my typo, I meant checked exception. Just make `zeroComparisonWithPositiveKeysTest() throws WrongStreamKeyException`. Again,  `assertDoesNotThrow` makes sense in the context of soft asserts only.